soc: mspm0: add power management support

SanjayyyV opened the pull request:
TI MSPM0 series supports range of power modes (RUN/SLEEP, STOP, STANDBY)
supporting low power operations. Provides automatic restoration to
RUN mode on wakeup from any low power state.

Signed-off-by: Sanjay Vallimanalan <sanjay@linumiz.com>

SanjayyyV committed:
dts: arm/ti/mspm0: add cpu power states for MSPM0 series

add range of cpu power states (run/sleep, stop and standby) for MSPM0
series. add exit latencies ranging from 1.5-15.7µs and minimum
residency times of 5-10ms for optimal power management transitions.

Signed-off-by: Sanjay Vallimanalan <sanjay@linumiz.com>
Signed-off-by: Parthiban Nallathambi <parthiban@linumiz.com>

SanjayyyV committed:
soc: mspm0: add power management support

TI MSPM0 series supports range of power modes (RUN/SLEEP, STOP, STANDBY)
supporting low power operations. Provides automatic restoration to
RUN mode on wakeup from any low power state.

Signed-off-by: Sanjay Vallimanalan <sanjay@linumiz.com>
Signed-off-by: Parthiban Nallathambi <parthiban@linumiz.com>

SanjayyyV committed:
soc: mspm0: add poweroff support

add support for SHUTDOWN operating mode in TI MSPM0 series for power-off
operation. Uses HWINFO for reset cause detection to handle shutdown IO
release on low power wakeup.

Signed-off-by: Sanjay Vallimanalan <sanjay@linumiz.com>
Signed-off-by: Parthiban Nallathambi <parthiban@linumiz.com>

parthitce commented on soc/ti/mspm0/common/power.c:
```diff
@@ -0,0 +1,83 @@
+{
+	switch (state) {
+	case DL_SYSCTL_POWER_POLICY_RUN_SLEEP0:
+		DL_SYSCTL_setPowerPolicyRUN0SLEEP0();
```
alignment is not correct. ditto for the whole file

parthitce reviewed (changes requested):
commit order needs to changed. dts needs to be the last after the code IMO

SanjayyyV committed:
drivers: gpio: Add GPIO Fast Wake support

The fast wake feature in the MSPM0 GPIO peripheral allows the GPIO module
to stay in a low-power state and detect interrupt events on the device pins
without requiring a high-speed clock. This allows the device to support
fast wakeup from low-power modes, such as STOP and STANDBY, on any GPIO
pin.

Signed-off-by: Sanjay Vallimanalan <sanjay@linumiz.com>
Signed-off-by: Parthiban Nallathambi <parthiban@linumiz.com>

SanjayyyV committed:
west.yml: Update hal_ti revision

Update hal_ti to include input-enable property for all UART RX pins.
Without the input-enable property, UART reception does not function
correctly when the SoC enters low-power mode. Adding this property
ensures reliable UART operation in all power states.

Signed-off-by: Sanjay Vallimanalan <sanjay@linumiz.com>

parthitce commented on soc/ti/mspm0/common/poweroff.c:
```diff
@@ -0,0 +1,40 @@
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/poweroff.h>
```
re-order to end

parthitce reviewed (changes requested):
Apart from the nit, changes are fine.

parthitce reviewed (approved):
LGTM
